src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1493,7 +1493,9 @@ tx_gateway_frontend::do_commit_tm_tx(
               group.group_id, group.etag, tx.pid, tx.tx_seq, timeout));
         }
 
-        if (tx.status == tm_transaction::tx_status::ongoing) {
+        if (
```

Comment:
I'm trying to understand how this plays with #6252..

What is the point of writing prepare (L1479-L1493) if we use acks=all? Is that coming in a different patch or I'm missing something?